exporter/datadogexporter/internal/metadata/internal/k8s/nodename.go

```diff
@@ -0,0 +1,63 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig"
+
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+type nodeNameProvider interface {
+	NodeName(context.Context) (string, error)
+}
+
+var _ nodeNameProvider = (*nodeNameProviderImpl)(nil)
+
+type nodeNameProviderImpl struct {
+	logger *zap.Logger
+	client k8s.Interface
+}
+
+func (p *nodeNameProviderImpl) namespace() string {
+	namespacePath := "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+	val, err := ioutil.ReadFile(namespacePath)
+	if err == nil && val != nil {
+		return string(val)
+	}
+	p.logger.Warn("Could not fetch k8s namespace, using 'default'", zap.Error(err))
+	return "default"
+}
+
+func (p *nodeNameProviderImpl) NodeName(ctx context.Context) (string, error) {
+	namespace := p.namespace()
+	podName, err := os.Hostname()
+	if err != nil {
+		return "", fmt.Errorf("could not fetch our hostname: %w", err)
+	}
+
+	pod, err := p.client.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
```

Comment:
should probably document the required rbac rules.

Reply:
It turns out the RBAC rules we currently document for supporting the k8sattributes processor are sufficient for this, so I just noted in the code that these should be kept in sync with the ones on our docs/examples